### PR TITLE
style: remove Space font style

### LINKED ISSUE #40315: Space component sets a font-size
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-zubuvo?file=%2Findex.js)

### Steps to reproduce

Nest a `Space` component inside a `Typography.Title` component.

### What is expected?

The font size of `Typography.Title` is respected.

### What is actually happening?

`Space` overrides the font size to be the default (`14px`)

| Environment | Info |
| --- | --- |
| antd | 5.1.5 |
| React | 18.2.0 |
| System | macOS |
| Browser | Firefox |

---
`Space` uses `genComponentStyleHook`, which in turn uses `genCommonStyle`, which then sets a `font-size` property on `Space` items like so:

```
:where(.css-dev-only-do-not-override-wcexds)[class^="ant-space"],
:where(.css-dev-only-do-not-override-wcexds)[class*=" ant-space"] {
 font-family:"Open Sans",sans-serif;
 font-size:14px;
 box-sizing:border-box;
}
```

This should not happen, spacing should not change font sizes. This might affect other components as well like `Row` or `Col` etc.

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #40315

### 💡 Background and solution

图片

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Space should not affect font style and font family.          |
| 🇨🇳 Chinese |  修复 Space 影响父元素字体大小和样式的问题。        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
https://preview-40318-ant-design.surge.sh/components/config-provider-cn#components-config-provider-demo-locale

image

是同一个问题吗大佬？

**Comment by li-jia-nan** [comment]
image

好像不是同一个问题，我这个问题注释掉 Demo 中的 key 就不会抖动了

**Comment by li-jia-nan** [comment]
argos diff 有点多，大佬 rebase 一下

**Review by li-jia-nan — APPROVED** [review]

**Comment by afc163** [comment]
这个需要周会讨论一下。

File: components/style/index.ts [hunk 1]
```diff
@@ -104,10 +104,18 @@ export const genCommonStyle = (token: DerivativeToken, componentPrefixCls: strin
 
   const rootPrefixSelector = `[class^="${componentPrefixCls}"], [class*=" ${componentPrefixCls}"]`;
 
+  // Space component don't apply extra font style
+  // https://github.com/ant-design/ant-design/issues/40315
+  const fontStyle = componentPrefixCls.endsWith(`-space`)
```
Comment (MadCcc) [inline components/style/index.ts:109]: 不如在 genComponentStyle 里加个配置项吧，把整个 commonStyle 都去掉

Comment (afc163) [reply components/style/index.ts:109]: ci 挂了，这个 componentPrefixCls 打印出来都是 `check`， @MadCcc 看看？

图片

**Pull request merged**